### DISCUSSION TITLE
Formula dependent on specific fork of library: possible?

### DISCUSSION THREAD

**Discussion by NJAldwin** [Homebrew/discussions#21] [Everyday usage]
I've been using the tool `kafkacat` recently: https://github.com/edenhill/kafkacat

One particular feature in `kafkacat` (Avro decoding + JSON envelope) requires using edenhill's fork of yajl (a JSON library): https://github.com/edenhill/yajl

The current formula for `kafkacat` depends on yajl, but the regular library, not the fork:
https://github.com/Homebrew/homebrew-core/blob/aba5c04012b253654b28fa7aa0620ad165306416/Formula/kafkacat.rb#L19

If I wanted to change the formula to depend on edenhill's fork of yajl, is that possible?  What would be required -- would I be able to (/should I) create a separate formula e.g. `edenhill-yajl` that installs the yajl fork, then depend on that?

fwiw, I am looking to contribute changes here back to the master formula (so kafkacat works out of the box for everybody), not on a custom tap.

**Comment by Rylan12** [comment]
I don't know if we allow forks of other formulae, so another maintainer should weigh in on that. If so, then yes, adding it and changing the `depends_on "yakl"` line to depend on the fork would probably do the job.

Instead of adding the `yajl` fork as a new formula, another option might be to add the `yajl` fork in a `resource` block.

If the change isn't accepted, you are encouraged to host your own [tap](https://docs.brew.sh/Taps) which can contain a formula for the `yajl` fork and a version of `kafkacat` that depends on it. More information about creating and maintaining a tap can be found [here](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap).

---

Edit: I didn't read this close enough so here's a better answer:

Yes, adding a new formula with the `yajl` tap is probably the best option, but I'm not positive it will be accepted because it's a fork of another formula.

If it's not accepted, another option is to add the `yajl` fork as a `resource` block to the `kafkacat` formula.

**Reply by SMillerDev** [reply]
To add on this, why doesn't the fork just upstream it's changes so everyone can work on the same codebase?

**Reply by NJAldwin** [reply]
It seems that the upstream `yajl` library has been abandoned: https://github.com/edenhill/kafkacat/issues/269#issuecomment-702306291

Perhaps that will bolster the case for making this fork available as a separate formula?

**Reply by SMillerDev** [reply]
If it's abandoned and then forked that would actually be a reason to change the original formula
